common/src/types/user.rs

```diff
@@ -10,3 +14,100 @@ pub struct Model {
     pub created_at: DateTime<Utc>,    // timestamptz
     pub last_login_at: DateTime<Utc>, // timestamptz
 }
+
+impl Model {
+    pub fn verify_password(&self, password: &str) -> bool {
+        let hash = match PasswordHash::new(&self.password_hash) {
+            Ok(hash) => hash,
+            Err(err) => {
+                tracing::error!("failed to parse password hash: {}", err);
+                return false;
+            }
+        };
+
+        Argon2::default()
+            .verify_password(password.as_bytes(), &hash)
+            .is_ok()
+    }
+}
+
+pub fn hash_password(password: &str) -> String {
+    let salt = SaltString::generate(&mut OsRng);
+
+    let hash = Argon2::default()
+        .hash_password(password.as_bytes(), &salt)
+        .expect("failed to hash password");
+
+    hash.to_string()
+}
```

Comment:
Consider adding peppering. Since argon2 (as an algo) even supports it with it's "secret"[^1], and `argon2` (the lib) implements it[^2]. Notes on how to properly do peppering from OWASP Password Cheat Sheet[^3], my beloved.

Also check out counter-points to pepper[^4] (and the two[^5] links[^6] from that answer are also interesting), even if **most of them don't apply to our case** because we are using argon2 as intended, it has the "secret" param for a reason. However, secret rotation should be considered, as OWASP[^3] also mentions.

Oh, and of course [my comment about passwords](https://github.com/ScuffleTV/scuffle/issues/36#issuecomment-1453244138) on #36 

[^1]: https://security.stackexchange.com/a/206177
[^2]: https://docs.rs/argon2/0.5.0/argon2/struct.Argon2.html ([method](https://docs.rs/argon2/0.5.0/argon2/struct.Argon2.html#method.new_with_secret))
[^3]: https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#peppering
[^4]: https://stackoverflow.com/a/16896216
[^5]: https://stackoverflow.com/a/16597402
[^6]: http://blog.ircmaxell.com/2012/04/properly-salting-passwords-case-against.html

Reply:
Peppering is annoying to work with and also does not really do much. Since peppering means we have to introduce state (we have to store the pepper) and if an attacker somehow manages to breach into our system, its very likely that they will have the pepper too.

Reply:
It just becomes annoying to work with since we then have to know about the pepper here. perhaps we can add it but I am not sure if it actually does anything extra. Argon2 is already impossible to crack using any means, adding peppering means we have to keep track of a state all for it to be `impossible * 2?`

Reply:
Maybe in the future, right now we can do without peppering.
Also, I'm reposting your comments to the issue
- #68